### PR TITLE
MA-285 Apply for outright possession warrant

### PR DESCRIPTION
**Pull request by maysakanoni**
**What?**

- Add  apply for outright possession warrant classification

**Why?**

- This needs to be flagged following a court outcome of 'OUT'

### CODE REVIEW

File: lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb [hunk 1]
```diff
@@ -43,12 +45,22 @@ def validate_wanted_action(wanted_action)
...
+          @criteria.court_outcome == 'OUT'
+        end
+
         def review_failed_letter?
           return false if @documents.empty?
           @documents.most_recent.failed? && @documents.most_recent.income_collection?
         end
 
         def send_informal_agreement_breach_letter?
+          return false if @criteria.balance.blank?
```
Comment (richrace) [inline lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb]: If this is mean to be present for all rules can we refactor it to be at the top of the `#execute`?

File: lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb [hunk 2]
```diff
@@ -43,12 +45,22 @@ def validate_wanted_action(wanted_action)
           raise ArgumentError, "Tried to classify a case as #{wanted_action}, but this is not on the list of valid classifications."
         end
 
+
+        def apply_for_outright_possession_warrant?
+          return false if @criteria.active_agreement?
+          return false if @criteria.courtdate.blank?
+          return false if @criteria.courtdate.future?
+          return false if @criteria.courtdate < 3.months.ago
+          @criteria.court_outcome == 'OUT'
```
[the remark this answers is not in the document]
Comment (maysakanoni) [reply lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb]: yes 3cd7bea40ff2bab1c9b13fc5806bd6d5f1cf7bc1

File: lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb (hunk 1 above)
Comment (maysakanoni) [reply lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb]: This isn't currently true for all the classifications

File: app/models/hackney/income/models/case_priority.rb [hunk 3]
```diff
@@ -5,7 +5,8 @@ class CasePriority < ApplicationRecord
         enum classification: {
           no_action: 0, send_letter_two: 1, send_letter_one: 2, send_first_SMS: 3, send_NOSP: 4,
           apply_for_court_date: 6, send_court_warning_letter: 7, update_court_outcome_action: 8,
-          send_court_agreement_breach_letter: 9, send_informal_agreement_breach_letter: 10
+          send_court_agreement_breach_letter: 9, send_informal_agreement_breach_letter: 10,
+          apply_for_outright_possession_warrant: 11
```
Comment (richrace) [inline app/models/hackney/income/models/case_priority.rb]: A merge conflict should highlight that this enum number might already be taken.

**Pull request merged**